tchannel-core/src/main/java/com/uber/tchannel/handlers/RequestRouter.java

```diff
@@ -251,11 +251,14 @@ public void onFailure(Throwable e) {
             private void doRequestEndProcessing() {
                 if (span != null) {
                     span.finish();
+                    TracingContext tracingContext = topChannel.getTracingContext();
```

Comment:
thank you for the review and your insight!
we also feel like thread-local context propagation in async flows is rather fragile and requires proper setup and participation from all moving parts.
going to merge the pr and publish new artifacts.